src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -0,0 +1,331 @@
+from __future__ import annotations  # for type hints
+
+from abc import ABC, abstractmethod
+from ast import parse as parse_python
+from collections.abc import Callable
+from enum import Enum
+
+from sqlglot import ParseError as SQLParseError
+from sqlglot import parse as parse_sql
+from databricks.sdk.service.workspace import Language
+
+NOTEBOOK_HEADER = " Databricks notebook source"
+CELL_SEPARATOR = " COMMAND ----------"
+MAGIC_PREFIX = ' MAGIC'
+LANGUAGE_PREFIX = ' %'
+
+
+class Cell(ABC):
+
+    def __init__(self, source: str):
+        self._original_code = source
+
+    @property
+    def migrated_code(self):
+        return self._original_code  # for now since we're not doing any migration yet
+
+    @property
+    @abstractmethod
+    def language(self) -> CellLanguage:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def is_runnable(self) -> bool:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        raise NotImplementedError()
+
+
+class PythonCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.PYTHON
+
+    def is_runnable(self) -> bool:
+        try:
+            ast = parse_python(self._original_code)
+            return ast is not None
+        except SyntaxError:
+            return False
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        pass  # TODO
+        # see https://github.com/databrickslabs/ucx/issues/1200
+        # see https://github.com/databrickslabs/ucx/issues/1202
```

Comment:
```suggestion
        # TODO https://github.com/databrickslabs/ucx/issues/1200
        # TODO https://github.com/databrickslabs/ucx/issues/1202
        pass
```

small nit - i'm planning to create issues from TODO automatically and let's have them follow the same'ish convention